lib/auth/usertoken.go

```diff
@@ -376,3 +393,55 @@ func (s *Server) getResetPasswordToken(ctx context.Context, tokenID string) (typ
 
 	return token, nil
 }
+
+// createRecoveryToken creates a user token for account recovery.
+func (s *Server) createRecoveryToken(ctx context.Context, username, tokenType string, usage types.UserTokenUsage) (types.UserToken, error) {
+	// TODO lisa add RecoveryApprovedToken type
+	if tokenType != UserTokenTypeRecoveryStart {
+		return nil, trace.BadParameter("invalid recovery token type")
+	}
+
+	if usage != types.UserTokenUsage_RECOVER_2FA && usage != types.UserTokenUsage_RECOVER_PWD {
```

Comment:
Check that `usage != _UNSPECIFIED`? It is alright otherwise, no?

Reply:
For this token, it has to be specified, and it has to be either `_MFA` or `_PASSWORD`